tests/test_discoverer.py

```diff
@@ -42,61 +72,81 @@ def test_discover_newly_copied_file():
         ],
         capture_output=True,
         text=True,
-        cwd="tests",
+        cwd=cwd,
     )
     db_select_result.check_returncode()
-    assert db_select_result.stdout[:-1] == os.path.basename(destination_file)
+    assert db_select_result.stdout[:-1] == destination_file
     assert db_select_result.stderr == ""
     # Teardown
     os.remove(file_path + destination_file)
-    db_delete_result = subprocess.run(
-        [
-            "sqlite3",
-            db_name,
-            "delete from files where id={};".format(expected_file_db_id),
-        ],
-        capture_output=True,
-        text=True,
-        cwd="tests",
-    )
-    db_delete_result.check_returncode()
+    teardown
 
 
 # Delete media file from monitored path
 def test_discover_removed_file():
     # Setup
+    setup
     shutil.copy2(file_path + origin_file, file_path + destination_file)
-    discoverer_test = discoverer_run_once()
-    discoverer_test.start()
-    assert int(max_db_id()) == 2
+    discoverer_run_once().start()
+    assert int(get_max_id_db()) == 2
+    assert int(get_num_entries_db()) == 2
     # Test
     os.remove(file_path + destination_file)
-    discoverer_test = discoverer_run_once()
-    discoverer_test.start()
-    assert int(max_db_id()) == 1
+    discoverer_run_once().start()
+    assert int(get_max_id_db()) == 1
+    assert int(get_num_entries_db()) == 1
+    # Teardown
+    teardown
 
 
 # Trigger manual rescan
 def test_manual_rescan():
     # Setup
+    setup
     shutil.copy2(file_path + origin_file, file_path + destination_file)
     process_handle = subprocess.Popen(flumes_discoverer, shell=True)
-    time.sleep(6)
+    time.sleep(3)
     os.rename(file_path + destination_file, file_path + modify_file)
     # Test
     subprocess.run("kill -USR1 " + str(process_handle.pid), shell=True)
-    time.sleep(6)
+    time.sleep(3)
     db_select_result = subprocess.run(
         ["sqlite3", db_name, "select name from files where id=2;"],
```

Comment:
Here it's best to replace the hardcoded value of 2 with a variable, see lines 63 and 71